router/broker.go

```diff
@@ -898,3 +1010,123 @@ func (b *broker) subCountSubscribers(msg *wamp.Invocation) wamp.Message {
 		Arguments: wamp.List{count},
 	}
 }
+
+// syncGetSubHistoryEvents returns all stored events for subscription
+// TODO: Need to filter by authid and/or authrole of caller if stored events have any of them
+// But that's not possible in current arch as meta peer doesn't know anything about caller, just invocation message
+func (b *broker) syncGetSubHistoryEvents(subId wamp.ID) []storedEvent {
+	if subscription, ok := b.subscriptions[subId]; ok {
+		if storeItem, ok := b.eventHistoryStore[subscription]; ok {
+			var events []storedEvent
```

Comment:
I meant something like:
```
events := make([]storedEvent, 0, len(storeItem.events))
for _, event := range storeItem.events {
    events = append(events, event.event)
}
```

This way you avoid re-allocation mid way through appending.

Reply:
Oh yeah! Thnx!